### PR TITLE
Additional make_network_data.py functionality

### PR DESCRIPTION
**Pull request by ff-l**

### CODE REVIEW

**Comment by ff-l** [comment]
Closes https://github.com/alphagov/govuk-network-data/issues/61

**Comment by ff-l** [comment]
Closes https://github.com/alphagov/govuk-network-data/issues/60

File: src/data/make_dataset.py [hunk 1]
```diff
@@ -52,7 +52,7 @@
 BATCH_SIZE: int = 3
 # A bit of a magic number, but limits dataframes that can be passed off to workers. If dataframe exceeds this size,
```
Comment (mammykins) [inline src/data/make_dataset.py]: this comment is vestigial? as data is no longer split and passed to workers

File: src/data/make_dataset.py [hunk 2]
```diff
@@ -177,7 +190,7 @@ def add_loop_columns(user_journey_df):
 
     # Count occurrences of de-looped journeys, most generic journey frequency metric.
     logger.debug("Aggregating de-looped journey occurrences...")
-    user_journey_df['Occurrences_NL'] = user_journey_df.groupby('Page_Seq_NL')['Page_Seq_Occurrences'].transform('sum')
+    user_journey_df['Occurrences_NL'] = user_journey_df.groupby('Page_Seq_NL')['Occurrences'].transform('sum')
```
Comment (mammykins) [inline src/data/make_dataset.py]: This was why the edge weights were too big!

File: src/data/make_network_data.py [hunk 3]
```diff
@@ -1,19 +1,20 @@
 import argparse
+import gzip
```
Comment (mammykins) [inline src/data/make_network_data.py]: is there any consensus on ordering? is it alphabetical or, when it's used?

Comment (ff-l) [reply src/data/make_network_data.py]: Not sure, you can "optimize imports" in pycharm though

**Comment by ff-l** [comment]
Closes https://github.com/alphagov/govuk-network-data/issues/66.

File: src/data/make_network_data.py [hunk 4]
```diff
@@ -1,66 +1,113 @@
...
 import preprocess as prep
 
-
-COLUMNS_TO_KEEP = ['Page_List_NL', 'Page_List', 'Occurrences', 'Page_Seq_Occurrences', 'Occurrences_NL']
+COLUMNS_TO_KEEP = ['Page_List', 'Page_List_NL', 'PageSequence', 'Page_Seq_NL', 'Occurrences', 'Page_Seq_Occurrences',
+                   'Occurrences_NL']
 
 
-def read_file(filename):
+def read_file(filename, use_delooped_journeys=False, drop_incorrect_occ=False):
```
Comment (mammykins) [inline src/data/make_network_data.py]: README to be updated in a later PR?

File: src/data/merge_dataset.py [hunk 5]
```diff
@@ -0,0 +1,344 @@
...
+                               'TrafficMediums', 'NetworkLocations', 'Dates']
+# Execute module for only one file
+SINGLE: bool = False
+# Fewer files to process than available cpus.
+FEWER_THAN_CPU: bool = False
+# Drop journeys occurring once (not in a day, multiple days, governed by DEPTH globals). If false, overrides depth
+# globals and keeps journeys, resulting in massive dataframes (danger zone).
+DROP_ONE_OFFS: bool = False
+# Drop journeys of length 1
+DROP_ONES: bool = False
```
Comment (mammykins) [inline src/data/merge_dataset.py]: This is repeated

File: src/data/preprocess.py [hunk 6]
```diff
@@ -23,12 +24,13 @@ def bq_journey_to_pe_list(bq_journey_string):
     :return: The list of page-event tuples.
     """
     # TODO: fix line below, for now it replaces string in a weird /search query within journey (uncommon)
-    bq_journey_string = bq_journey_string.replace(">>iii....", "")
+    bq_journey_string = re.sub(">>iii\.+|>>\.+|\s>>>\s", "", bq_journey_string)
```
Comment (mammykins) [inline src/data/preprocess.py]: maybe a comment here to explain why we need this

File: src/data/queries/stnd_taxon.sql [hunk 7]
```diff
@@ -0,0 +1,67 @@
+SELECT
```
Comment (mammykins) [inline src/data/queries/stnd_taxon.sql]: assumed both queries tested (to save the cost of me running them separately)

File: src/data/merge_dataset.py [hunk 8]
```diff
@@ -0,0 +1,344 @@
...
+def list_to_dict(metadata_list):
+    """
+    Transform metadata lists to dictionary aggregates
+    :param metadata_list:
+    :return:
+    """
+    return Counter([xs for xs in metadata_list])
+
+
+def str_to_dict(metadata_str):
```
Comment (mammykins) [inline src/data/merge_dataset.py]: alot of this has already been reviewed and tested in early PRs?

**Comment by ff-l** [comment]
:shipit:

**Pull request merged**